File: xNet/src/main/java/org/sqlunet/provider/XNetContract.java
/*
 * Copyright (c) 2022. Bernard Bou
 */

package org.sqlunet.provider;

import org.sqlunet.loaders.Queries;
import org.sqlunet.xnet.provider.Q;
import org.sqlunet.xnet.provider.V;

/**
 * Extended cross WordNet-FrameNet-PropBank-VerbNet provider contract
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public final class XNetContract
{
	// A L I A S E S

	public static final String AS_WORDS = V.AS_WORDS;
	public static final String AS_SENSES = V.AS_SENSES;
	public static final String AS_SYNSETS = V.AS_SYNSETS;
	public static final String AS_POSES = V.AS_POSES;
	public static final String AS_CLASSES = V.AS_CLASSES;

	public interface Words_FnWords_PbWords_VnWords
	{
		String URI = "words_fnwords_pbwords_vnwords";
		String WORD = V.WORD;
		String WORDID = V.WORDID;
		String FNWORDID = V.FNWORDID;
		String PBWORDID = V.PBWORDID;
		String VNWORDID = V.VNWORDID;
		String SYNSETID = V.SYNSETID;
		String LUID = V.LUID;
		String SENSEID = V.SENSEID;
		String SENSENUM = V.SENSENUM;
		String SENSEKEY = V.SENSEKEY;
		String POSID = V.POSID;
		String POS = V.POS;
		String DOMAIN = V.DOMAIN;
		String DEFINITION = V.DEFINITION;
		String CASED = V.CASEDWORD;
		String TAGCOUNT = V.TAGCOUNT;
		String SOURCES = V.SOURCES;
	}

	public interface Words_Pronunciations_FnWords_PbWords_VnWords extends Words_FnWords_PbWords_VnWords
	{
		String URI = "words_pronunciations_fnwords_pbwords_vnwords";
		String PRONUNCIATION = V.PRONUNCIATION;
		String VARIETY = V.VARIETY;
		String PRONUNCIATIONS = Queries.PRONUNCIATIONS;
	}

	public interface Words_PbWords_VnWords
	{
		String URI = "words_pbwords_vnwords";
		String WORD = V.WORD;
		String WORDID = V.WORDID;
		String PBWORDID = V.PBWORDID;
		String VNWORDID = V.VNWORDID;
		String SYNSETID = V.SYNSETID;
		String LUID = V.LUID;
		String SENSEID = V.SENSEID;
		String SENSENUM = V.SENSENUM;
		String SENSEKEY = V.SENSEKEY;
		String POSID = V.POSID;
		String POS = V.POS;
		String DOMAIN = V.DOMAIN;
		String DEFINITION = V.DEFINITION;
		String CASEDWORD = V.CASEDWORD;
		String TAGCOUNT = V.TAGCOUNT;
		String SOURCES = V.SOURCES;
	}

	public interface Words_VnWords_VnClasses
	{
		String URI = "words_vnwords_vnclasses";
		String WORDID = V.WORDID;
		String SYNSETID = V.SYNSETID;
		String VNWORDID = V.VNWORDID;
		String CLASSID = V.CLASSID;
		String CLASS = V.CLASS;
		String CLASSTAG = V.CLASSTAG;
		String DEFINITION = V.DEFINITION;
		String SOURCE = V.SOURCE;
		String SOURCES = V.SOURCES;
	}

	public interface Words_PbWords_PbRoleSets
	{
		String URI = "words_pbwords_pbrolesets";
		String WORDID = V.WORDID;
		String SYNSETID = V.SYNSETID;
		String ROLESETID = V.ROLESETID;
		String ROLESETNAME = V.ROLESETNAME;
		String ROLESETHEAD = V.ROLESETHEAD;
		String ROLESETDESCR = V.ROLESETDESCR;
		String DEFINITION = V.DEFINITION;
		String SOURCE = V.SOURCE;
		String SOURCES = V.SOURCES;
	}

	public interface Words_XNet
	{
		String WORDID = V.WORDID;
		String SYNSETID = V.SYNSETID;
		String XID = V.XID;
		String XCLASSID = V.XCLASSID;
		String XMEMBERID = V.XMEMBERID;
		String XNAME = V.XNAME;
		String XHEADER = V.XHEADER;
		String XINFO = V.XINFO;
		String XDEFINITION = V.XDEFINITION;
		String SOURCE = V.SOURCE;
		String SOURCES = V.SOURCES;
	}

	// Sources

	public interface Sources
	{
		String TABLE = Q.SOURCES.TABLE;
		String URI = Sources.TABLE;
		String ID = "idsource";
		String NAME = "name";
		String VERSION = "version";
		String URL = "url";
		String PROVIDER = "provider";
		String REFERENCE = "reference";
	}

	// Meta

	public interface Meta
	{
		String TABLE = Q.META.TABLE;
		String URI = Meta.TABLE;
		String CREATED = "created";
		String DBSIZE = "dbsize";
		String BUILD = "build";
	}

	// PredicateMatrix

	public interface PredicateMatrix
	{
		String URI = "predicatematrix";
		String WORDID = V.WORDID;
		String SYNSETID = V.SYNSETID;
		String VNWORDID = V.VNWORDID;
		String VNCLASSID = V.CLASSID;
		String PBWORDID = V.PBWORDID;
		String PBROLESETID = V.ROLESETID;
		String FNWORDID = V.FNWORDID;
		String LUID = V.LUID;
		String FRAMEID = V.FRAMEID;
	}

	public interface PredicateMatrix_VerbNet
	{
		String URI = "predicatematrix_verbnet";
		String WORDID = V.WORDID;
		String SYNSETID = V.SYNSETID;
		String VNWORDID = V.VNWORDID;
		String CLASSID = V.CLASSID;
	}

	public interface PredicateMatrix_PropBank
	{
		String URI = "predicatematrix_propbank";
		String WORDID = V.WORDID;
		String SYNSETID = V.SYNSETID;
		String PBWORDID = V.PBWORDID;
		String ROLESETID = V.ROLESETID;
	}

	public interface PredicateMatrix_FrameNet
	{
		String URI = "predicatematrix_framenet";
		String WORDID = V.WORDID;
		String SYNSETID = V.SYNSETID;
		String FNWORDID = V.FNWORDID;
		String LUID = V.LUID;
		String FRAMEID = V.FRAMEID;
	}

	// PredicateMatrix unions

	public interface Words_XNet_U
	{
		String WORDID = V.WORDID;
		String SYNSETID = V.SYNSETID;
		String XID = V.XID;
		String XCLASSID = V.XCLASSID;
		String XMEMBERID = V.XMEMBERID;
		String XNAME = V.XNAME;
		String XHEADER = V.XHEADER;
		String XINFO = V.XINFO;
		String XPRONUNCIATION = V.XPRONUNCIATION;
		String XDEFINITION = V.XDEFINITION;
		String SOURCE = V.SOURCE;
		String SOURCES = V.SOURCES;
	}

	public interface Words_VnWords_VnClasses_U
	{
		String URI = "words_vnwords_vnclasses_u";
		String WORDID = V.WORDID;
		String SYNSETID = V.SYNSETID;
		String VNWORDID = V.VNWORDID;
		String CLASSID = V.CLASSID;
		String CLASS = V.CLASS;
		String CLASSTAG = V.CLASSTAG;
		String DEFINITION = V.DEFINITION;
		String SOURCE = V.SOURCE;
		String SOURCES = V.SOURCES;
	}

	public interface Words_VnWords_VnClasses_1U2
	{
		String URI = "words_vnwords_vnclasses_1u2";
	}

	public interface Words_VnWords_VnClasses_1
	{
		String URI = "words_vnwords_vnclasses_1";
	}

	public interface Words_VnWords_VnClasses_2
	{
		String URI = "words_vnwords_vnclasses_2";
	}

	public interface Words_PbWords_PbRoleSets_U
	{
		String URI = "words_pbwords_pbrolesets_u";
		String WORDID = V.WORDID;
		String SYNSETID = V.SYNSETID;
		String ROLESETID = V.ROLESETID;
		String ROLESETNAME = V.ROLESETNAME;
		String ROLESETHEAD = V.ROLESETHEAD;
		String ROLESETDESCR = V.ROLESETDESCR;
		String DEFINITION = V.DEFINITION;
		String SOURCE = V.SOURCE;
		String SOURCES = V.SOURCES;
	}

	public interface Words_PbWords_PbRoleSets_1
	{
		String URI = "words_pbwords_pbrolesets_1";
	}

	public interface Words_PbWords_PbRoleSets_2
	{
		String URI = "words_pbwords_pbrolesets_2";
	}

	public interface Words_PbWords_PbRoleSets_1U2
	{
		String URI = "words_pbwords_pbrolesets_1u2";
	}

	public interface Words_FnWords_FnFrames_U
	{
		String URI = "words_fnwords_fnframes_u";
		String WORDID = V.WORDID;
		String SYNSETID = V.SYNSETID;
		String FNWORDID = V.FNWORDID;
		String FRAMEID = V.FRAMEID;
		String FRAME = V.FRAME;
		String FRAMEDEFINITION = V.FRAMEDEFINITION;
		String LUID = V.LUID;
		String LEXUNIT = V.LEXUNIT;
		String LUDEFINITION = V.LUDEFINITION;
		String DEFINITION = V.DEFINITION;
		String SOURCE = V.SOURCE;
		String SOURCES = V.SOURCES;
	}

	public interface Words_FnWords_FnFrames_1U2
	{
		String URI = "words_fnwords_fnframes_1u2";
	}

	public interface Words_FnWords_FnFrames_1
	{
		String URI = "words_fnwords_fnframes_1";
	}

	public interface Words_FnWords_FnFrames_2
	{
		String URI = "words_fnwords_fnframes_2";
	}
}

File: xNet/src/main/java/org/sqlunet/provider/XNetContract.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.provider

import org.sqlunet.loaders.Queries
import org.sqlunet.xnet.provider.Q
import org.sqlunet.xnet.provider.V

/**
 * Extended cross WordNet-FrameNet-PropBank-VerbNet provider contract
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
object XNetContract {

    // A L I A S E S

    const val AS_WORDS = V.AS_WORDS
    const val AS_SENSES = V.AS_SENSES
    const val AS_SYNSETS = V.AS_SYNSETS
    const val AS_POSES = V.AS_POSES
    const val AS_CLASSES = V.AS_CLASSES

    object Words_FnWords_PbWords_VnWords {
        const val URI = "words_fnwords_pbwords_vnwords"
        const val WORD = V.WORD
        const val WORDID = V.WORDID
        const val FNWORDID = V.FNWORDID
        const val PBWORDID = V.PBWORDID
        const val VNWORDID = V.VNWORDID
        const val SYNSETID = V.SYNSETID
        const val LUID = V.LUID
        const val SENSEID = V.SENSEID
        const val SENSENUM = V.SENSENUM
        const val SENSEKEY = V.SENSEKEY
        const val POSID = V.POSID
        const val POS = V.POS
        const val DOMAIN = V.DOMAIN
        const val DEFINITION = V.DEFINITION
        const val CASED = V.CASEDWORD
        const val TAGCOUNT = V.TAGCOUNT
        const val SOURCES = V.SOURCES
    }

    object Words_Pronunciations_FnWords_PbWords_VnWords {
        const val URI = "words_pronunciations_fnwords_pbwords_vnwords"
        const val PRONUNCIATION = V.PRONUNCIATION
        const val VARIETY = V.VARIETY
        const val PRONUNCIATIONS = Queries.PRONUNCIATIONS

        const val WORD = Words_FnWords_PbWords_VnWords.WORD
        const val WORDID = Words_FnWords_PbWords_VnWords.WORDID
        const val FNWORDID = Words_FnWords_PbWords_VnWords.FNWORDID
        const val PBWORDID = Words_FnWords_PbWords_VnWords.PBWORDID
        const val VNWORDID = Words_FnWords_PbWords_VnWords.VNWORDID
        const val SYNSETID = Words_FnWords_PbWords_VnWords.SYNSETID
        const val LUID = Words_FnWords_PbWords_VnWords.LUID
        const val SENSEID = Words_FnWords_PbWords_VnWords.SENSEID
        const val SENSENUM = Words_FnWords_PbWords_VnWords.SENSENUM
        const val SENSEKEY = Words_FnWords_PbWords_VnWords.SENSEKEY
        const val POSID = Words_FnWords_PbWords_VnWords.POSID
        const val POS = Words_FnWords_PbWords_VnWords.POS
        const val DOMAIN = Words_FnWords_PbWords_VnWords.DOMAIN
        const val DEFINITION = Words_FnWords_PbWords_VnWords.DEFINITION
        const val CASED = Words_FnWords_PbWords_VnWords.CASED
        const val TAGCOUNT = Words_FnWords_PbWords_VnWords.TAGCOUNT
        const val SOURCES = Words_FnWords_PbWords_VnWords.SOURCES
    }

    object Words_PbWords_VnWords {
        const val URI = "words_pbwords_vnwords"
        const val WORD = V.WORD
        const val WORDID = V.WORDID
        const val PBWORDID = V.PBWORDID
        const val VNWORDID = V.VNWORDID
        const val SYNSETID = V.SYNSETID
        const val LUID = V.LUID
        const val SENSEID = V.SENSEID
        const val SENSENUM = V.SENSENUM
        const val SENSEKEY = V.SENSEKEY
        const val POSID = V.POSID
        const val POS = V.POS
        const val DOMAIN = V.DOMAIN
        const val DEFINITION = V.DEFINITION
        const val CASEDWORD = V.CASEDWORD
        const val TAGCOUNT = V.TAGCOUNT
        const val SOURCES = V.SOURCES
    }

    object Words_VnWords_VnClasses {
        const val URI = "words_vnwords_vnclasses"
        const val WORDID = V.WORDID
        const val SYNSETID = V.SYNSETID
        const val VNWORDID = V.VNWORDID
        const val CLASSID = V.CLASSID
        const val CLASS = V.CLASS
        const val CLASSTAG = V.CLASSTAG
        const val DEFINITION = V.DEFINITION
        const val SOURCE = V.SOURCE
        const val SOURCES = V.SOURCES
    }

    object Words_PbWords_PbRoleSets {
        const val URI = "words_pbwords_pbrolesets"
        const val WORDID = V.WORDID
        const val SYNSETID = V.SYNSETID
        const val ROLESETID = V.ROLESETID
        const val ROLESETNAME = V.ROLESETNAME
        const val ROLESETHEAD = V.ROLESETHEAD
        const val ROLESETDESCR = V.ROLESETDESCR
        const val DEFINITION = V.DEFINITION
        const val SOURCE = V.SOURCE
        const val SOURCES = V.SOURCES
    }

    object Words_XNet {
        const val WORDID = V.WORDID
        const val SYNSETID = V.SYNSETID
        const val XID = V.XID
        const val XCLASSID = V.XCLASSID
        const val XMEMBERID = V.XMEMBERID
        const val XNAME = V.XNAME
        const val XHEADER = V.XHEADER
        const val XINFO = V.XINFO
        const val XDEFINITION = V.XDEFINITION
        const val SOURCE = V.SOURCE
        const val SOURCES = V.SOURCES
    }

    // Sources

    object Sources {
        const val TABLE = Q.SOURCES.TABLE
        const val URI = TABLE
        const val ID = "idsource"
        const val NAME = "name"
        const val VERSION = "version"
        const val URL = "url"
        const val PROVIDER = "provider"
        const val REFERENCE = "reference"
    }

    // Meta

    object Meta {
        const val TABLE = Q.META.TABLE
        const val URI = TABLE
        const val CREATED = "created"
        const val DBSIZE = "dbsize"
        const val BUILD = "build"
    }

    // PredicateMatrix

    object PredicateMatrix {
        const val URI = "predicatematrix"
        const val WORDID = V.WORDID
        const val SYNSETID = V.SYNSETID
        const val VNWORDID = V.VNWORDID
        const val VNCLASSID = V.CLASSID
        const val PBWORDID = V.PBWORDID
        const val PBROLESETID = V.ROLESETID
        const val FNWORDID = V.FNWORDID
        const val LUID = V.LUID
        const val FRAMEID = V.FRAMEID
    }

    object PredicateMatrix_VerbNet {
        const val URI = "predicatematrix_verbnet"
        const val WORDID = V.WORDID
        const val SYNSETID = V.SYNSETID
        const val VNWORDID = V.VNWORDID
        const val CLASSID = V.CLASSID
    }

    object PredicateMatrix_PropBank {
        const val URI = "predicatematrix_propbank"
        const val WORDID = V.WORDID
        const val SYNSETID = V.SYNSETID
        const val PBWORDID = V.PBWORDID
        const val ROLESETID = V.ROLESETID
    }

    object PredicateMatrix_FrameNet {
        const val URI = "predicatematrix_framenet"
        const val WORDID = V.WORDID
        const val SYNSETID = V.SYNSETID
        const val FNWORDID = V.FNWORDID
        const val LUID = V.LUID
        const val FRAMEID = V.FRAMEID
    }

// PredicateMatrix unions

    object Words_XNet_U {
        const val WORDID = V.WORDID
        const val SYNSETID = V.SYNSETID
        const val XID = V.XID
        const val XCLASSID = V.XCLASSID
        const val XMEMBERID = V.XMEMBERID
        const val XNAME = V.XNAME
        const val XHEADER = V.XHEADER
        const val XINFO = V.XINFO
        const val XPRONUNCIATION = V.XPRONUNCIATION
        const val XDEFINITION = V.XDEFINITION
        const val SOURCE = V.SOURCE
        const val SOURCES = V.SOURCES
    }

    object Words_VnWords_VnClasses_U {
        const val URI = "words_vnwords_vnclasses_u"
        const val WORDID = V.WORDID
        const val SYNSETID = V.SYNSETID
        const val VNWORDID = V.VNWORDID
        const val CLASSID = V.CLASSID
        const val CLASS = V.CLASS
        const val CLASSTAG = V.CLASSTAG
        const val DEFINITION = V.DEFINITION
        const val SOURCE = V.SOURCE
        const val SOURCES = V.SOURCES
    }

    object Words_VnWords_VnClasses_1U2 {
        const val URI = "words_vnwords_vnclasses_1u2"
    }

    object Words_VnWords_VnClasses_1 {
        const val URI = "words_vnwords_vnclasses_1"
    }

    object Words_VnWords_VnClasses_2 {
        const val URI = "words_vnwords_vnclasses_2"
    }

    object Words_PbWords_PbRoleSets_U {
        const val URI = "words_pbwords_pbrolesets_u"
        const val WORDID = V.WORDID
        const val SYNSETID = V.SYNSETID
        const val ROLESETID = V.ROLESETID
        const val ROLESETNAME = V.ROLESETNAME
        const val ROLESETHEAD = V.ROLESETHEAD
        const val ROLESETDESCR = V.ROLESETDESCR
        const val DEFINITION = V.DEFINITION
        const val SOURCE = V.SOURCE
        const val SOURCES = V.SOURCES
    }

    object Words_PbWords_PbRoleSets_1 {
        const val URI = "words_pbwords_pbrolesets_1"
    }

    object Words_PbWords_PbRoleSets_2 {
        const val URI = "words_pbwords_pbrolesets_2"
    }

    object Words_PbWords_PbRoleSets_1U2 {
        const val URI = "words_pbwords_pbrolesets_1u2"
    }

    object Words_FnWords_FnFrames_U {
        const val URI = "words_fnwords_fnframes_u"
        const val WORDID = V.WORDID
        const val SYNSETID = V.SYNSETID
        const val FNWORDID = V.FNWORDID
        const val FRAMEID = V.FRAMEID
        const val FRAME = V.FRAME
        const val FRAMEDEFINITION = V.FRAMEDEFINITION
        const val LUID = V.LUID
        const val LEXUNIT = V.LEXUNIT
        const val LUDEFINITION = V.LUDEFINITION
        const val DEFINITION = V.DEFINITION
        const val SOURCE = V.SOURCE
        const val SOURCES = V.SOURCES
    }

    object Words_FnWords_FnFrames_1U2 {
        const val URI = "words_fnwords_fnframes_1u2"
    }

    object Words_FnWords_FnFrames_1 {
        const val URI = "words_fnwords_fnframes_1"
    }

    object Words_FnWords_FnFrames_2 {
        const val URI = "words_fnwords_fnframes_2"
    }
}

